Reduce bandwidth at low zoom level: 'trick' the map into showing lower zoom levels.

garvankeeley opened the pull request:
For example, if the map is at zoom level x, show the tiles for x - 4. I don't do this exactly.

This code has a low/high zoom level switch, at
the high zoom level, behave as before this commit; at the low zoom
level, trick the map into showing very low resolution tiles, so only a
few tiles need loading to cover the map area.

garvankeeley committed:
Reduce badnwidth at low zoom level: 'trick' the map into showing lower zoom levels.
For example, if the map is at zoom level x, show the tiles for x - 4.
I don't do this exactly.
This code has a low/high zoom level switch, at
the high zoom level, behave as before this commit; at the low zoom
level, trick the map into showing very low resolution tiles, so only a
few tiles need loading to cover the map area.

garvankeeley commented:
Video demonstrating this feature: 
https://drive.google.com/file/d/0B1sYPwjczMsQWGtFSUdLTTZKS0U

First is high-res mode, it shows the lower resolution coverage map when zoomed out, and a higher res coverage when zoomed in. Next is low-res mode, showing two levels of low res map tiles, depending on the zoom level.  Notice the lower right label on the map is the status.

crankycoder commented on android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java:
```diff
@@ -74,16 +78,19 @@
     private static final int DEFAULT_ZOOM_AFTER_FIX = 16;
     private static final String LAT_KEY = "latitude";
     private static final String LON_KEY = "longitude";
+    private static final int HIGH_LOW_ZOOM_THRESHOLD = 14;
```
Rename to HIGH_ZOOM_THRESHOLD to be consistent with https://github.com/mozilla/MozStumbler/pull/998/files#diff-ecc6477a0a55adf1f37c8d8de23f5f85R284

garvankeeley replied:
Nevermind, I see, change the name to HIGH_ZOOM_THRESHOLD

crankycoder commented on android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapActivity.java:
```diff
@@ -239,38 +264,77 @@ public void run() {
+        return zoomLevel > HIGH_LOW_ZOOM_THRESHOLD;
+    }
+
+    private void updateOverlayBaseLayer(int zoomLevel) {
```
Can you put a comment in this method similar to `updateOverlayCoverageLayer` so that it's clear what we're trying to do here.

garvankeeley committed:
rename HIGH_ZOOM_THRESHOLD

garvankeeley committed:
Merge branch 'dev' of https://github.com/garvankeeley/MozStumbler into low-res-tiles

Rename MapActivity to MapFragment

garvankeeley committed:
Move the network receiver to its own class, and bind its lifetime
clearly to the lifetime of the MapFragment onStart onStop. Add lots of
guards against the activity disappearing on us.

garvankeeley commented:
Argh. I used git-cola to push, and 9a2c66e has the wrong commit message. Let me fix...

Actually, no it is ok, the main thing is "Rename MapActivity to MapFragment", so it is correct.

garvankeeley committed:
Merge branch 'dev' of https://github.com/garvankeeley/MozStumbler into 2-level-low-res

Conflicts:
	android/src/main/java/org/mozilla/mozstumbler/client/mapview/MapFragment.java
	android/src/main/java/org/mozilla/mozstumbler/client/navdrawer/MainDrawerActivity.java
